Ci testing

BenediktSeidl opened the pull request.

TheSpacyCat committed:
extend scaffolded webpack config with EnvironmentPlugin

This way, we can behave differently depending on Environment Variables at build time.

TheSpacyCat committed:
use build variable to modify behaviour for cloud edition plugin

BenediktSeidl committed:
split github actions: build should be reusable

this way we can execute a whole build on each commit.

BenediktSeidl committed:
fix validator: move screenshot to repo

README.md needs absolute paths, but plugin.json a relative...

BenediktSeidl committed:
fix validator: fix badge

BenediktSeidl committed:
build plugin twice: signed and unsigned

BenediktSeidl committed:
add validate plugin step in ci

BenediktSeidl committed:
update node to match @grafana/create-plugin

TheSpacyCat committed:
fix another rebranding leftover

TheSpacyCat commented on .github/workflows/ci.yml:
```diff
@@ -62,6 +62,31 @@ jobs:
+        run: |
+          git clone https://github.com/grafana/plugin-validator
+          pushd ./plugin-validator/pkg/cmd/plugincheck2
+          go install
```
We can use a binary from one of the releases. The tars contain an executable file, a README and a LICENSE, nothing else.

BenediktSeidl replied:
this was c&p from the offical actions, but you are right, i adapted it.

TheSpacyCat replied:
fine for me

BenediktSeidl replied:
does not work, as this file `./plugin-validator/config/default.yaml` is missing. i'm going to remove this commit. we can look into this later

TheSpacyCat commented on README.md:
```diff
@@ -1,6 +1,6 @@
 # Checkmk's Grafana Data Source Plugin
 
-![CI](https://github.com/checmk/grafana-checkmk-datasource/actions/workflows/ci.yml/badge.svg)
+![CI](https://github.com/Checkmk/grafana-checkmk-datasource/actions/workflows/ci.yml/badge.svg)
```
Ok, I'll have to rewrite these everywhere else as well. I'll put it on my TODO list.

BenediktSeidl replied:
are you sure? i can not finy any matches for this typo.

TheSpacyCat replied:
Found a few other ones in the plugin.json. I'll do a fixup MR

TheSpacyCat commented on .github/build/action.yml:
```diff
@@ -0,0 +1,76 @@
+    required: false
+    default: ''
+outputs:
+  changelog_outputs_path:
```
Maybe we'll give these nice names in the future, e.g. changelog, archive, checksum

BenediktSeidl replied:
should be done

TheSpacyCat commented on .github/build/action.yml:
```diff
@@ -0,0 +1,76 @@
+  steps:
+    - name: Install dependencies
+      shell: bash
+      run: yarn install --frozen-lockfile
```
Just out of interest: What is the difference between '--immutable' and '--forzen-lockfile'

BenediktSeidl replied:
i have no idea. this was c&p from the grafana provided workflow. but they changed from yarn to npm and in the current workflows, the both use `npm ci` already added the switch to npm on our long term todo

TheSpacyCat replied:
fine for me

TheSpacyCat commented on .github/workflows/ci.yml:
```diff
@@ -32,17 +32,21 @@ jobs:
+        id: build
+        uses: ./.github/build/
+        with:
+          build_identifier: dev.${{ github.run_id }}
```
Have you thought about using a short commit sha here?

BenediktSeidl replied:
we can if you want. but you can build twice for one commit, and the run id will increment (i hope)

TheSpacyCat replied:
fine for me.

BenediktSeidl replied:
would keept it as is, but we can adapt later on if you want.

TheSpacyCat commented on .github/workflows/release.yml:
```diff
@@ -76,9 +37,9 @@ jobs:
+          artifacts: ./${{ steps.build.metadata_outputs_archive }}
           artifactContentType: application/zip
 
       - name: Add checksum to release
```
We can put that into the step above.

BenediktSeidl replied:
but then they have both `artifactContentType: raw`. we will see if this is a problem

TheSpacyCat commented on .github/workflows/release.yml:
```diff
@@ -76,9 +37,9 @@ jobs:
+          body: ${{ steps.build.changelog_outputs_path }}
           draft: true
-          artifacts: ./${{ steps.metadata.outputs.archive }}
+          artifacts: ./${{ steps.build.metadata_outputs_archive }}
```
That one can be changed to './*zip' to include a zip for free and cloud.

BenediktSeidl replied:
i think it's good to be specific here, i would hope we can notice problems earlier this way.

TheSpacyCat commented on .github/build/action.yml:
```diff
@@ -70,6 +99,9 @@ runs:
       shell: bash
       id: package-plugin
       run: |
+        # clean up previous build # TODO: think about better solution
```
We could check out the repo two times and run the cloud build in one path and the free build in the other.

BenediktSeidl replied:
adapted this one of the commits

BenediktSeidl committed:
use checkout action to reset repo

BenediktSeidl committed:
rename plugins:

Title: Checkmk data source
plugin-id: tribe-29-checkmk-datasource

Title: Checkmk data source for Checkmk Cloud Edition
plugin-id: checkmk-checkmk-datasource

Also adapt headline in README.md

BenediktSeidl committed:
adapt release action
